server/events/vcs/github_client.go

```diff
@@ -709,6 +710,24 @@ pagination:
 	return reviewDecision, requiredChecks, requiredWorkflows, checkRuns, statusContexts, nil
 }
 
+// GetLatestCheckRun returns the checkRun with the highest runNumber, i.e., the latest checkRun whose status we need to evaluate.
+func GetLatestCheckRun(checkRuns []CheckRun) (CheckRun, error) {
+	latestCheckRunNumber := 0
+	for _, checkRun := range checkRuns {
+		if int(checkRun.CheckSuite.WorkflowRun.RunNumber) > latestCheckRunNumber {
+			latestCheckRunNumber = int(checkRun.CheckSuite.WorkflowRun.RunNumber)
+		}
+	}
+
+	for _, checkRun := range checkRuns {
```

Comment:
There is no need to iterate twice, the check run with the highest run number can be returned immediately after the first loop if it is kept as a variable pointer.